app/services/solidus_pay_tomorrow/client/base_service.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'httparty'
+
+module SolidusPayTomorrow
+  module Client
+    class BaseService
+      attr_reader :payment_method
+
+      STAGING_URL = 'https://api-staging.paytomorrow.com'
+      PRODUCTION_URL = 'https://api.paytomorrow.com'
+
+      def initialize(payment_method:, **_args)
+        @payment_method = payment_method
+      end
+
+      def self.call(*args, **kwargs, &block)
+        new(*args, **kwargs, &block).call
+      end
+
+      private
+
+      def api_base_url
+        if payment_method.preferred_test_mode
+          STAGING_URL
+        else
+          PRODUCTION_URL
+        end
+      end
+
+      def auth_headers
+        { 'Authorization': "Bearer #{ENV['ACCESS_TOKEN']}",
+          'Content-Type': 'application/json' }
+      end
+
+      def handle_errors(result)
+        return result.parsed_response if result.success?
+
+        raise StandardError, result.parsed_response['error']
```

Comment:
Considering this method is raising an expection, what do you think adding a `!` at the end of the method name?

Reply:
Added